File: Auth/Auth.App/Interface/RefreshToken/IRefreshTokenRepository.cs
namespace Auth.App.Interface.RefreshToken
{
    public interface IRefreshTokenRepository
    {
        Task<RefreshTokenRecord?> GetAsync(string token);
        Task CreateAsync(RefreshTokenRecord record);
        Task UpdateAsync(RefreshTokenRecord record);
        Task DeleteAsync(string token);
        Task DeleteAsync(Guid userId);
        void Migrate();
    }
}

File: NodeAuth/src/repos/interfaces/IRefreshTokenRepository.ts
import { RefreshTokenRecord } from "../../models/RefreshTokenRecord";

export interface IRefreshTokenRepository {
  getAsync(token: string): Promise<RefreshTokenRecord | null>;
  createAsync(record: RefreshTokenRecord): Promise<void>;
  updateAsync(record: RefreshTokenRecord): Promise<void>;
  deleteAsync(token: string): Promise<void>;
  deleteByUserIdAsync(userId: string): Promise<void>;
  migrate(): void;
}
